cli/azd/pkg/containerapps/container_app.go

```diff
@@ -49,22 +47,20 @@ type ContainerAppService interface {
 // NewContainerAppService creates a new ContainerAppService
 func NewContainerAppService(
 	credentialProvider account.SubscriptionCredentialProvider,
-	httpClient httputil.HttpClient,
 	clock clock.Clock,
+	defaultClientOptionsBuilder *azsdk.ClientOptionsBuilderFactory,
 ) ContainerAppService {
 	return &containerAppService{
-		credentialProvider: credentialProvider,
-		httpClient:         httpClient,
-		userAgent:          azdinternal.UserAgent(),
-		clock:              clock,
+		credentialProvider:          credentialProvider,
+		clock:                       clock,
+		defaultClientOptionsBuilder: defaultClientOptionsBuilder,
```

Comment:
My biggest concern is these call-time variables like `subscriptionId` in this case and, in almost all other cases, `ctx`. If we can accept these from the container then we can, as you point out, inject fully hydrated clients.

Reply:
Met w/ @chlowell yesterday. A few takeaways: 

* `ctx` should not be bound to policy (we're not binding the `ctx` object directly to the policy, rather we're using it to transport the trace id)... instead we should use the `ctx` at call time (instead of creation time as we're doing today). This amounts to changing the policy to use the context available in the `Do` func. 
* The design of binding `subscriptionId` to an Azure SDK client at creation is in accordance with the accepted patterns. (@wbreza and I also discussed that we think the `subscriptionId` is unlikely to change during a single execution of `azd`) 

To that end I'm going to: 
* Refactor `correlation_policy.go` to use the ctx from the request at call-time 
* Create a type for `subscriptionId` which can be populated using the IoC container (and other types as needed) 

This should enable the injection of hydrated clients. Updates coming soon.